src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+        _registerOperatorWithCoordinator(msg.sender, quorumNumbers, pubkey);
+    }
+
+    /**
+     * @notice Registers msg.sender as an operator with the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registering for
+     * @param pubkey is the BLS public key of the operator
+     */
+    function registerOperatorWithCoordinator(
+        bytes calldata quorumNumbers, 
+        BN254.G1Point memory pubkey,
+        OperatorKickParam[] calldata operatorKickParams
+    ) external {
+        require(quorumNumbers.length == operatorKickParams.length, "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: quorumNumbers and operatorKickParams must be the same length");
+        // register the operator
+        _registerOperatorWithCoordinator(msg.sender, quorumNumbers, pubkey);
+
+        // get the registering operator's operatorId
+        bytes32 registeringOperatorId = _operators[msg.sender].operatorId;
+
+        // kick the operators
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            // check that the quorum has reached the max operator count
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            OperatorSetParam memory operatorSetParam = quorumOperatorSetParams[quorumNumber];
+            {
+                uint32 numOperatorsForQuorum = indexRegistry.totalOperatorsForQuorum(quorumNumber);
+                require(
+                    numOperatorsForQuorum == operatorSetParam.maxOperatorCount + 1,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: quorum has not reached max operator count"
+                );
+
+                // get the total stake for the quorum
+                uint96 totalStakeForQuorum = stakeRegistry.getCurrentTotalStakeForQuorum(quorumNumber);
+                bytes32 operatorToKickId = _operators[operatorKickParams[i].operator].operatorId;
+                uint96 operatorToKickStake = stakeRegistry.getCurrentOperatorStakeForQuorum(operatorToKickId, quorumNumber);
+                uint96 registeringOperatorStake = stakeRegistry.getCurrentOperatorStakeForQuorum(registeringOperatorId, quorumNumber);
+
+                // check the registering operator has more than the kick percentage of the operator to kick's stake
+                require(
+                    registeringOperatorStake > operatorToKickStake * operatorSetParam.kickPercentageOfOperatorStake / 100,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: registering operator has less than kickPercentageOfOperatorStake"
+                );
+                
+                // check that the operator to kick has less than the kick percentage of the average stake
+                require(
+                    operatorToKickStake < totalStakeForQuorum * operatorSetParam.kickPercentageOfAverageStake / 100 / numOperatorsForQuorum,
```

Comment:
this math syntax makes me uncomfortable. maybe at least have a parentheses before dividing a second time?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/61db48095b57b7ea6b0fe66cb87061c72ce31d9c